### PR TITLE
Include imageonly attribute for repositories

### LINKED ISSUE #362: Set repos just for 'imageinclude' not for build time
## Description

Add the possibility to define repositories to be included into the image which are not actually used during build time.

This issues is included to discuss and track the latest comments from here #178

### PR DESCRIPTION
**Pull request by davidcassany**
This commit adds imageonly attribute support for the repository
element. imageonly is a boolean attribute that if true indicates
that the repository is not used for the build but needs to be
configured for the resulting image.

Fixes #362

### CODE REVIEW

File: kiwi/system/prepare.py [hunk 1]
```diff
@@ -99,7 +99,7 @@ def setup_repositories(self, clear_cache=False, signing_keys=None):
         :rtype: PackageManager
         """
         repository_options = []
-        repository_sections = self.xml_state.get_repository_sections()
+        repository_sections = self.xml_state.get_build_repository_sections()
```
Comment (schaefi) [inline kiwi/system/prepare.py]: Could we rename this to

```python
get_repository_sections_used_for_build()
```

File: kiwi/system/setup.py [hunk 2]
```diff
@@ -115,7 +115,7 @@ def import_repositories_marked_as_imageinclude(self):
         imageinclude attribute should be permanently added to
         the image repository configuration
         """
-        repository_sections = self.xml_state.get_repository_sections()
+        repository_sections = self.xml_state.get_image_repository_sections()
```
Comment (schaefi) [inline kiwi/system/setup.py]: Could we rename this to

```python
get_repository_sections_used_in_image()
```

Comment (schaefi) [inline kiwi/system/setup.py]: Reason for the name change is that I was confused what an image_repository is. With the background of the existing imageinclude and imageonly attributes it's clear but I think we could make it more explicit by a name change of the functions. What do you think ?

**Comment by schaefi** [comment]
Looks great to me, nice python coding :) I should do more of these too. Just a minor issue with the function names. Let me know what you think

File: kiwi/system/setup.py (hunk 2 above)
Comment (davidcassany) [inline kiwi/system/setup.py]: Yes, that looks great to me, I am not very good at naming stuff... I'll change that, I'm totally fine with more explicit names.

**Pull request merged**